### PR TITLE
Unit test

### PR DESCRIPTION
**Pull request by karam084**
**Project requirements**

- Add unit tests to your classes:
1. Book
2. Person
3. Student
4. Teacher
5. Classroom
6. Rental
7. all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require_relative '../person'
+
+describe Person do
+  it 'create a new person with a first and last name' do
+    person = Person.new(20, 'Smith', parent_permission: true)
+    expect(person.name).to eq('Smith')
+    expect(person.age).to eq(20)
+    expect(person.can_use_services?).to eql true
```
Comment (youmari) [inline spec/person_spec.rb:8]: - [x] it would be better if you add a test for correct name method in order to cover all the methods

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @karam084 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb (hunk 1 above)
Comment (aneallaryea100) [reply spec/person_spec.rb:8]: correct name method added to test to cover all method

**Review by sumairq — APPROVED** [review]
### Hi Team, 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**